cmd/migration-tool/main.go

```diff
@@ -0,0 +1,524 @@
+	}
+
+	go mainProcessLoop(ctxWithCancel, stopCh, imageClient.ImageV1(), releasePayloadClient.ReleaseV1alpha1(), o.Namespace, o.Name, o.ProwNamespace, o.Execute, o.Location)
+
+	select {
+	case <-ctx.Done():
+		klog.V(5).Infof("Context has been cancelled")
+	case <-stopCh:
+		klog.V(5).Infof("mainProcessLoop returned")
+	}
+	return nil
+}
+
+func loadClusterConfig(overrides *clientcmd.ConfigOverrides) (*rest.Config, error) {
+	cfg := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(clientcmd.NewDefaultClientConfigLoadingRules(), overrides)
+	clusterConfig, err := cfg.ClientConfig()
+	if err != nil {
+		return nil, fmt.Errorf("could not load client configuration: %v", err)
+	}
+	return clusterConfig, nil
+}
+
+func mainProcessLoop(ctx context.Context, stopCh chan bool, imageClient imageclient.ImageV1Interface, releasePayloadClient releasev1alpha1.ReleaseV1alpha1Interface, namespace, name, prowNamespace string, execute bool, location string) {
+	ctxWithTimeout, cancelFunction := context.WithTimeout(ctx, time.Duration(15)*time.Second)
+
+	defer func() {
+		cancelFunction()
+		stopCh <- true
+	}()
+
+	var errors []error
+	var payloads []*v1alpha1.ReleasePayload
+	controller := newController(ctxWithTimeout, imageClient, namespace, name, prowNamespace)
+
+	if !execute {
+		klog.Infof("Running in Dry-Run mode.  No objects will be created on cluster.")
+	}
+
+	if len(location) > 0 {
+		payloads = readPayloads(location)
+
+		if execute {
+			errors = createPayloads(releasePayloadClient, payloads)
+		}
+	} else {
+		releaseConfigs, err := controller.processImagestreams()
+		if err != nil {
+			klog.Errorf("Unable to gather release configs: %v", err)
+			return
+		}
+		releases, skipped, problems := controller.processReleaseImagestream()
+		payloads = controller.generatePayloads(releases, releaseConfigs)
+
+		klog.Infof("Successfully generated %d ReleasePayload definitions", len(payloads))
+		klog.Warningf("The following %d tag(s) have problems:\n%s", len(problems), strings.Join(problems, ","))
+		klog.Warningf("The following %d tag(s) cannot be migrated:\n%s", len(skipped), strings.Join(skipped, ","))
+
+		if execute {
+			errors = createPayloads(releasePayloadClient, payloads)
+		} else {
+			writePayloads(payloads)
+		}
+	}
+
+	if len(errors) > 0 {
+		klog.Info("The following errors occurred:\n")
+		for _, e := range errors {
+			klog.Warningf(e.Error())
+		}
+	}
+}
+
+type Controller struct {
+	Context       context.Context
+	ImageClient   imageclient.ImageV1Interface
+	Namespace     string
+	Name          string
+	ProwNamespace string
+}
+
+func newController(ctx context.Context, client imageclient.ImageV1Interface, namespace, name, prowNamespace string) *Controller {
+	return &Controller{
+		Context:       ctx,
+		ImageClient:   client,
+		Namespace:     namespace,
+		Name:          name,
+		ProwNamespace: prowNamespace,
+	}
+}
+
+func (c Controller) processImagestreams() (map[string]*releasecontroller.ReleaseConfig, error) {
```

Comment:
if the goal is for this to be a controller in the future, I suggest having this method take args for imagestreamnamespace and imagestreamname, so that you can wire to an informer and a workqueue later.